packages/beacon-node/src/api/impl/validator/index.ts

```diff
@@ -490,196 +499,184 @@ export function getValidatorApi({
-          ? // TODO deneb: builderSelection needs to be figured out if to be done beacon side
-            // || builderSelection !== BuilderSelection.BuilderOnly
-            produceEngineFullBlockOrContents(slot, randaoReveal, graffiti, {
-              feeRecipient,
-              strictFeeRecipientCheck,
-              // skip checking and recomputing head in these individual produce calls
-              skipHeadChecksAndUpdate: true,
-              commonBlockBody,
-            }).catch((e) => {
-              logger.error("produceEngineFullBlockOrContents failed to produce block", {slot}, e);
-              return null;
-            })
-          : null;
-
-      let blindedBlock, fullBlock;
-      if (blindedBlockPromise !== null && fullBlockPromise !== null) {
-        // reference index of promises in the race
-        const promisesOrder = [ProducedBlockSource.builder, ProducedBlockSource.engine];
-        [blindedBlock, fullBlock] = await racePromisesWithCutoff<
-          | ((routes.validator.ProduceBlockOrContentsRes | routes.validator.ProduceBlindedBlockRes) & {
-              shouldOverrideBuilder?: boolean;
-            })
-          | null
-        >(
-          [blindedBlockPromise, fullBlockPromise],
-          BLOCK_PRODUCTION_RACE_CUTOFF_MS,
-          BLOCK_PRODUCTION_RACE_TIMEOUT_MS,
-          // Callback to log the race events for better debugging capability
-          (event: RaceEvent, delayMs: number, index?: number) => {
-            const eventRef = index !== undefined ? {source: promisesOrder[index]} : {};
-            logger.verbose("Block production race (builder vs execution)", {
-              event,
-              ...eventRef,
-              delayMs,
-              cutoffMs: BLOCK_PRODUCTION_RACE_CUTOFF_MS,
-              timeoutMs: BLOCK_PRODUCTION_RACE_TIMEOUT_MS,
-              slot,
-            });
-          }
-        );
-        if (blindedBlock instanceof Error) {
-          // error here means race cutoff exceeded
-          logger.error("Failed to produce builder block", {slot}, blindedBlock);
-          blindedBlock = null;
+      const builderDisabledError = new Error("Builder disabled");
+      const engineDisabledError = new Error("Engine disabled");
+      const [builder, engine] = await resolveOrRacePromises(
+        [
+          isBuilderEnabled
+            ? // can't do fee recipient checks as builder bid doesn't return feeRecipient as of now
+              produceBuilderBlindedBlock(slot, randaoReveal, graffiti, {
+                feeRecipient,
+                // skip checking and recomputing head in these individual produce calls
+                skipHeadChecksAndUpdate: true,
+                commonBlockBody,
+              })
+            : Promise.reject<ReturnType<typeof produceBuilderBlindedBlock>>(builderDisabledError),
+
+          isEngineEnabled // TODO deneb: builderSelection needs to be figured out if to be done beacon side
```

Comment:
isEngineEnabled includes `builderSelection !== BuilderSelection.BuilderOnly` so this todo can be removed?

Reply:
I tried not to change any comment or log message, but yes it can be removed.